src/tic_tac_clojure/cli.clj

```diff
@@ -0,0 +1,25 @@
+(ns tic-tac-clojure.cli
+  (:gen-class)
+  (:require [tic-tac-clojure.message-render :refer :all]
+            [tic-tac-clojure.utils :refer :all]
+            [tic-tac-clojure.validators :refer :all]))
+
+(defn print-to-cli
+  [message]
+  (println (str message)))
+            
+(defn extract-numeric-input
+  []
+  (let [input (read-line)]
+  (if (is-number? input)
+      (convert-to-num input)
+      (do (print-to-cli not-a-number-message)
+          (recur)))))    
+            
+(defn get-valid-num-input 
```

Comment:
Add a couple of tests for this function to ensure that it prompts for a valid number when it gets an invalid number, and that it returns a valid number. If you're not sure how to test that it recurs, remember that you can nest calls to with-out-str and with-in-str

Reply:
I've now written tests for this function.